File: pset2/Vigenere/vigenere.c
#include <stdlib.h>
#include <string.h>
#include <stdio.h>
#include <ctype.h>
#include <cs50.h>

int main(int argc, string argv[])
{
    int k_length;
    string k;


    while (argc!=2)
    {
     printf("Error: ./vigenere string\n");
     return 1;
    }

    k=argv[1];
    k_length = strlen(argv[1]);
    int j=0;

    for(int i=0; i<k_length; i++)
    {
        if(!isalpha(k[i]))
            {
                printf("Error: ./vigenere string\n");
                return 1;
            }
    }


    string s=get_string("plaintext: "); //s=plaintext

    printf("ciphertext: ");


    for (int i=0; i<strlen(s); i++)
        {
            if (isalpha(s[i]))
                {
                    int key;
                    key=tolower(k[j%k_length]) - 'a';

                    if (isupper(s[i]))
                        {

                                 printf("%c",( (s[i]+key) - 65) % 26 + 65);
                        }

                       if (islower(s[i]))
                        {
                                printf("%c", ((s[i]+key) - 97) % 26 + 97);

                        }

                    j++;
                }

            else
                {
                    printf("%c", s[i]);
                }
        }


    printf("\n");
    return 0;
}

File: pset6/vigenere/vigenere.py
from cs50 import get_string
import sys
def main():
    if len(sys.argv) != 2:
        print("Usage: ./vigenere string")
        sys.exit(1)
    k=sys.argv[1]
    if(k.isalpha()==False):
        print("Usage: ./vigenere string")
        sys.exit(1)
    s=get_string("plaintext: ")
    print("ciphertext: ", end="")

    j=0
    for c in s:
        if(c.isalpha()):
            key=ord(k[j%len(k)].lower())-97
            if(c.isupper()):
                d=ord(c)
                output=chr((d-65+key)%26+65)
                j+=1
            elif(c.islower()):
                d=ord(c)
                output=chr((d-97+key)%26+97)
                j+=1
        else:
            output=c
        print(output,end="")
    print("")

if __name__=="__main__":
    main()